packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -240,19 +283,34 @@ export class DetailsService {
   private assignReputationsToLeaders(
     leaders: LeaderDto[],
     reputations: { address: string; reputation: string }[],
-    chainId: ChainId,
-  ) {
+    orderBy: LeadersOrderBy,
+    first: number,
+  ): LeaderDto[] {
+    const leaderMap = new Map(leaders.map((l) => [l.address.toLowerCase(), l]));
+
+    if (orderBy === LeadersOrderBy.REPUTATION) {
+      return reputations
+        .map((rep) => {
+          const leader = leaderMap.get(rep.address.toLowerCase());
+          if (leader) {
```

Comment:
@portuu3 @eugenvoronov can you think about such case? IMU it can't be that we have a reputation record but don't have leader address in chain. If it's not returned by subgraph then it's probably an issue and in this case I would:
- write a log that we don't have leader for reputation record
- skip this leader for response
- do not substitute this "missing leader" with another one

i.e. in code it will look like
```
reputations.slice(0, first)
  .map(/*existing logic*/)
  .filter(Boolean)
```

In any case - would be nice to drop a comment in the code about what & why so somebody new doesn't try to break it w/o full context

Reply:
You know, since we have `orderBy === LeadersOrderBy.REPUTATION ? first : undefined,`  in `this.fetchReputations`, I don't think we really need that `.slice(0, first)`.

Reply:
You know, since we have `orderBy === LeadersOrderBy.REPUTATION ? first : undefined`, in `this.fetchReputations`, I don't think we really need that `.slice(0, first)`.

Reply:
Good point! ☝🏻